### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by mbhall88**
## Added
- Editor integration instructions for PyCharm
- Instructions for adding badge
- CHANGELOG template from <https://keepachangelog.com/en/1.0.0/>
- Instructions for using configuration (`pyproject.toml`) file

### Changed
- Added a few examples to the example file
- Fixed weird tab/indentation in the example file code snippet

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,25 +6,33 @@
...
-This repository provides formatting for [Snakemake][snakemake] files. It follows the design and specifications of [Black][black].
+This repository provides formatting for [Snakemake][snakemake] files. It follows the
+design and specifications of [Black][black].
 
-> **⚠️WARNING⚠️**: As this project is still in the very early stages of development, and thus not stable, we strongly recommend ensuring your files are under version control before doing any formatting. Alternatively, you can pipe the file in from stdin, which will print it to the screen, or use the `--diff` option. See [Usage](#usage) for more details.
+> **⚠️WARNING⚠️**: As this project is still in the very early stages of development, and
+> thus not stable, we strongly recommend ensuring your files are under version control
+> before doing any formatting. Alternatively, you can pipe the file in from stdin, which
+> will print it to the screen, or use the `--diff` option. See [Usage](#usage) for more
+> details.
```
Comment (bricoletc) [inline README.md:17]: Given we're going to make a release, I'd like to replace 'As this project is still in the very early stages of development, and thus not stable' with 'snakefmt modifies files in-place by default, thus we strongly recommend...'. We could add a sentence after that saying 'this guards against any bugs as well' to stay safe, but the jist at this point is to warn that files will get modified in place

File: README.md [hunk 2]
```diff
@@ -34,6 +42,88 @@ cd snakefmt && poetry install
...
 ```
 
+## Example File
+
+Input
+
+```python
+from snakemake.utils import min_version
+min_version("5.14.0")
+configfile: "config.yaml" # snakemake keywords are treated like classes
```
Comment (bricoletc) [inline README.md:52]: I like the additions- here i would add 'treated like classes (2 newlines) maybe, so it's clear why we say this.

The second time you write this (after `include`) could just write` # 2 newlines `

File: README.md (hunk 1 above)
Comment (mbhall88) [reply README.md:17]: Yeah, agreed. I'll change this section.

File: README.md (hunk 2 above)
Comment (mbhall88) [reply README.md:52]: :+1:

**Pull request merged**